src/UserSpaceInstrumentation/FindFunctionAddress.cpp

```diff
@@ -8,30 +8,25 @@
 
 #include <string>
 
-#include "ModuleUtils/ReadLinuxModules.h"
 #include "ModuleUtils/VirtualAndAbsoluteAddresses.h"
 #include "ObjectUtils/ElfFile.h"
 
 namespace orbit_user_space_instrumentation {
 
-ErrorMessageOr<uint64_t> FindFunctionAddress(pid_t pid, std::string_view module_soname,
-                                             std::string_view function_name) {
-  auto modules = orbit_module_utils::ReadModules(pid);
-  if (modules.has_error()) {
-    return modules.error();
-  }
-
+ErrorMessageOr<uint64_t> FindFunctionAddress(
+    const std::vector<orbit_grpc_protos::ModuleInfo>& modules, std::string_view module_soname,
+    std::string_view function_name) {
   std::string module_file_path;
   uint64_t module_base_address = 0;
-  for (const orbit_grpc_protos::ModuleInfo& module : modules.value()) {
+  for (const orbit_grpc_protos::ModuleInfo& module : modules) {
     if (module.soname() == module_soname) {
       module_file_path = module.file_path();
       module_base_address = module.address_start();
     }
   }
   if (module_file_path.empty()) {
     return ErrorMessage(
-        absl::StrFormat("There is no module \"%s\" in process %d.", module_soname, pid));
+        absl::StrFormat("There is no module \"%s\" in the target process", module_soname));
```

Comment:
I removed the `pid` parameter from this function as now it was only used for this one message. As a result, I could also remove it from `FindFunctionAddressWithFallback`.